language/move-lang/src/typing/core.rs

```diff
@@ -1006,33 +1021,19 @@ fn solve_implicitly_copyable_constraint(
 ) {
     let ty = unfold_type(&context.subst, ty);
     let tloc = ty.loc;
-    let kind = match infer_kind(&context, &context.subst, ty.clone()) {
-        // Anything => None
-        // Unbound TVar or Anything satisfies any constraint. Will fail later in expansion
-        None => return,
-        Some(k) => k,
-    };
-    match kind {
-        sp!(_, Kind_::Copyable) => (),
-        sp!(kloc, Kind_::Affine) => {
-            let ty_str = error_format(&ty, &context.subst);
-            context.error(vec![
-                (loc, format!("{} {}", msg, fix)),
-                (tloc, format!("The type: {}", ty_str)),
-                (
-                    kloc,
-                    "Is declared as a non-implicitly copyable type here".into(),
+    if !is_implicitly_copyable(&context.subst, &ty) {
+        let ty_str = error_format(&ty, &context.subst);
+        context.error(vec![
+            (loc, format!("{} {}", msg, fix)),
+            (
+                tloc,
+                format!(
+                    "The type {} is not implicitly copyable. Implicit copies is limitied so \
```

Comment:
I'm going crazy on the typos, but you've got 3 wrong words in a row here: "is limitied so" should be "are limited to"